src/main/java/com/redhat/rhjmc/containerjfr/net/web/handlers/HealthGetHandler.java

```diff
@@ -0,0 +1,140 @@
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import com.google.gson.Gson;
+
+import com.redhat.rhjmc.containerjfr.core.log.Logger;
+import com.redhat.rhjmc.containerjfr.core.sys.Environment;
+import com.redhat.rhjmc.containerjfr.net.web.HttpMimeType;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+
+class HealthGetHandler implements RequestHandler {
+
+    static final String GRAFANA_DATASOURCE_ENV = "GRAFANA_DATASOURCE_URL";
+    static final String GRAFANA_DASHBOARD_ENV = "GRAFANA_DASHBOARD_URL";
+
+    private final Provider<CloseableHttpClient> httpClientProvider;
+    private final Environment env;
+    private final Gson gson;
+    private final Logger logger;
+
+    @Inject
+    HealthGetHandler(
+            Provider<CloseableHttpClient> httpClientProvider,
+            Environment env,
+            Gson gson,
+            Logger logger) {
+        this.httpClientProvider = httpClientProvider;
+        this.env = env;
+        this.gson = gson;
+        this.logger = logger;
+    }
+
+    @Override
+    public String path() {
+        return "/health";
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.GET;
+    }
+
+    // try-with-resources generates a "redundant" nullcheck in bytecode
+    @SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE")
+    @Override
+    public void handle(RoutingContext ctx) {
+        boolean datasourceAvailable = false;
+        boolean dashboardAvailable = false;
+
+        if (this.env.hasEnv(GRAFANA_DATASOURCE_ENV)) {
+            try (CloseableHttpResponse response =
+                    this.httpClientProvider
+                            .get()
+                            .execute(new HttpGet(this.env.getEnv(GRAFANA_DATASOURCE_ENV))); ) {
+                if (response.getStatusLine().getStatusCode() == 200) {
+                    datasourceAvailable = true;
+                }
+            } catch (IOException e) {
+                logger.warn(e);
+            }
+        }
+
+        if (this.env.hasEnv(GRAFANA_DASHBOARD_ENV)) {
+            String url = this.env.getEnv(GRAFANA_DASHBOARD_ENV) + "/api/health";
+            try (CloseableHttpResponse response =
+                    this.httpClientProvider.get().execute(new HttpGet(url)); ) {
+                if (response.getStatusLine().getStatusCode() == 200) {
+                    dashboardAvailable = true;
+                }
+            } catch (IOException e) {
+                logger.warn(e);
+            }
+        }
+
+        ctx.response()
+                .putHeader(HttpHeaders.CONTENT_TYPE, HttpMimeType.JSON.mime())
+                .end(
+                        gson.toJson(
+                                new TreeMap<>(
```

Comment:
Is the `TreeMap` here jjust to ensure consistent ordering of the keys?

Reply:
Yeah; let me know if there's a smarter way to do it D:

Reply:
Hmm. I don't know of a particularly better way, although I wonder if it really matters - a typical client parsing the JSON response shouldn't care about the key order. Maybe it makes testing easier, but then perhaps the tests should grab a Gson instance (from the MainModule), parse the response, and make assertions on what is found in the deserialized structure?

Reply:
Hey Andrew. I think that clients will appreciate consistent responses and the cost for us to do so is minimal, though a bit... hacky (here). If you'd like, I can keep it random and adjust the test, let me know.

Reply:
I don't really see the benefit to clients of having consistently ordered JSON keys, though. JSON parsing of an object should be independent of what order the keys appear in, so any client using a spec-compliant parser shouldn't even really notice whether the keys are sorted or even in consistent order or not. If we start enforcing a particular ordering of the keys here then a) we should be doing it everywhere and b) that ordering becomes an implicit part of our API, and I don't quite agree with either of those.

Reply:
Okay. Updated in the latest commit